src/main/java/io/r2dbc/proxy/callback/ReactiveBatchCallback.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.r2dbc.proxy.callback;
+
+import io.r2dbc.proxy.core.ConnectionInfo;
+import io.r2dbc.proxy.core.ExecutionType;
+import io.r2dbc.proxy.core.QueryExecutionInfo;
+import io.r2dbc.proxy.core.QueryInfo;
+import io.r2dbc.spi.Batch;
+import io.r2dbc.spi.Result;
+import org.reactivestreams.Publisher;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Proxy callback for {@link Batch}.
+ *
+ * @author Tadaya Tsuyukubo
+ */
+public class ReactiveBatchCallback extends CallbackSupport {
+
+    private Batch<?> batch;
+
+    private ConnectionInfo connectionInfo;
+
+    private List<String> queries = new ArrayList<>();
+
```

Comment:
Should this initialization be moved into the constructor call?

Reply:
Um, yes, it can, but isn't it more common to initialize collection at declaration if they are not assigned by constructor?